tempesta_fw/cache.c

```diff
@@ -660,7 +660,10 @@ tfw_cache_send_304(TfwHttpReq *req, TfwCacheEntry *ce)
 	if (tfw_msg_write(&it, &g_crlf))
 		goto err_setup;
 
-	tfw_http_resp_fwd(resp);
+	if (TFW_CONN_H2(req->conn))
+		tfw_h2_resp_adjust_fwd(resp);
+	else
+		tfw_http_resp_fwd(resp);
```

Comment:
OK. I inserted the TODOs with comments and stub functions in appropriate places. The actual implementation of that functions is performed in context of next PR - for HPACK (since this functionality is closely intertwined with HPACK encoding).

Reply:
>Cache is just a data storage - it's better to separate it from the transport layer, so it's better to move the TFW_CONN_H2(req->conn)-dependent logic to http*.[ch].
In this particular case this is problematic, since although for HTTP/2 we need separate (from HTTP/1.1) logic to assemble 304 response, but as with HTTP/1.1, we need to call our HTTP/2 procedures from cache logic (to HPACK indexing, encoding and writing data in-place).